modules/compute/gke-node-pool/outputs.tf

```diff
@@ -24,8 +24,10 @@ locals {
   last_digit  = trimsuffix(try(local.machine_vals[2], 0), "g")
 
   # Shared core machines only have 1 cpu allocatable, even if they have 2 cpu capacity
-  vcpu        = local.machine_shared_core ? 1 : local.is_a_series ? local.last_digit * 12 : local.last_digit
-  useable_cpu = local.set_threads_per_core ? local.threads_per_core * local.vcpu / 2 : local.vcpu
+  # Logic updated to work with TPUs - is vcpu a supported concept for TPUs as well?
```

Comment:
Added a temporary logic here. Need guidance on the vcpu calculation for TPUs. Also, the logic on line 29 seems to be considering a2 series machine and has a `* 12`. What about A3 and A4 series?